### PR TITLE
issue #1530 - Fix date selection in the SDP

### LINKED ISSUE #1530: SingleDatePicker doesn't fire onDateChange in IE11
**react-dates version**
e.g. react-dates@18.4.1

**Describe the bug**
In, IE11, SingleDatePicker renders and opens on focus. But when I click on a date, the calendar disappears without updating the date in the input. This works fine on other browsers.

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser IE
 - Version 11.0.9600.19236

**Is the issue reproducible in Storybook?**
https://airbnb.io/react-dates/?selectedKind=SingleDatePicker%20%28SDP%29&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

### PR DESCRIPTION
**Pull request by GattoLupesco**
Fixes #1530

### CODE REVIEW

File: src/components/SingleDatePicker.jsx [hunk 1]
```diff
@@ -248,7 +248,7 @@ class SingleDatePicker extends React.PureComponent {
 
   onFocusOut(e) {
     const { onFocusChange } = this.props;
-    if (this.dayPickerContainer.contains(e.relatedTarget || e.target)) return;
+    if (this.dayPickerContainer.contains(e.relatedTarget || e.target) || this.dayPickerContainer.contains(e.target)) return;
```
Comment (ljharb) [inline src/components/SingleDatePicker.jsx]: Can we add a test and/or story to cover this?

It seems like if `e.relatedTarget` is falsy, this line will cause `contains` to be invoked twice with the same `e.target`. Can that be avoided?

**Comment by jrparish** [comment]
@GattoLupesco - do you need help with this at all? This is a critical fix and we'd love to have your changes as soon as possible.

**Comment by ljharb** [comment]
@jrparish if you wanted to write tests, ping me here with a link to your branch and i can pull them into this PR.

**Comment by jrparish** [comment]
@ljharb - I made some tweaks to match the logic to what is being done in the `DateRangePicker` and then added some tests - 
Branch: https://github.com/jrparish/react-dates/tree/fix_singledatepicker_focus_on_ie11
Commit: https://github.com/jrparish/react-dates/commit/b75d56c91a5731f2cfb972872b7681e597a827d7

File: karma.conf.js [hunk 2]
```diff
@@ -36,7 +37,6 @@ module.exports = (config) => {
               presets: ['airbnb'],
             },
           },
-          { test: /\.json$/, loader: 'json-loader' },
```
Comment (ljharb) [inline karma.conf.js]: why is this removed?

**Comment by ljharb** [comment]
@jrparish Although the karma tests aren't required, they still failed here - is that something you'd be willing to take a look at? (in a followup PR would be fine too)

**Comment by ljharb** [comment]
actually i'm going to merge this as-is, since it leaves the karma tests in a much better situation.

File: karma.conf.js (hunk 2 above)
Comment (jrparish) [reply karma.conf.js]: This was throwing an error when running karma and is not needed with webpack 4.

**Comment by jrparish** [comment]
I can take a look at the other karma tests in a follow up 👍

**Pull request merged**